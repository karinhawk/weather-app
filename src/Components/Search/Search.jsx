import "./Search.scss"

const Search = ({captureInput, searchForm}) => {
  return (
    <div>
    <form onSubmit={captureInput}>
        <input ref={searchForm} className='searchbox' type="text" placeholder="Town... City... Country..." />
        <button className="form-button" type="submit">Go</button>
    </form>
    </div>
  )
}

export default Search