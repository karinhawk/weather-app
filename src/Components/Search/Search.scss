.searchbox {
    height: 5vh;
    width: 50vw;
    border: none;
    border-radius: 0.5rem;
    margin-right: 0.3rem;
}

.form-button {
    height: 5vh;
    width: 10vw;
    border: none;
    border-radius: 0.3rem;
}