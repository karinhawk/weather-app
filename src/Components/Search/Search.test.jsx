import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Search from "./Search.jsx";

it("Should render the searchbar", () => {
    const {asFragment} = render(<Search />)
    expect(asFragment(<Search />)).toMatchSnapshot()
  });

it("Should update input box with each key press", () => {
    render(<Search/>);
  
    const input = screen.getByPlaceholderText("Town... City... Country...");
    userEvent.type(input, "London");
  
    const updatedInput = screen.findByText("London");
    expect(updatedInput).toBeTruthy();
  })

it("Should not display the wrong order of characters based on what was typed", () => {
render(<Search/>);

const input = screen.getByPlaceholderText("Town... City... Country...");
userEvent.type(input, "St Ives");

const updatedInput = screen.queryByText("Ives St");
expect(updatedInput).toBeFalsy();
});