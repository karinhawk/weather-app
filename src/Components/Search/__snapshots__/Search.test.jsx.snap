// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the searchbar 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <input
      class="form__searchbox"
      placeholder="Town... City... Country..."
      type="text"
    />
    <button
      class="form__button"
      type="submit"
    >
      Go
    </button>
  </form>
</DocumentFragment>
`;
